man/ctags-optlib.7.rst.in

```diff
@@ -285,14 +294,110 @@ representation. ``--list-regex-flags`` lists all the flags.
 	manner.
 
 ``placeholder``
-	TBW
+	Don't emit a tag captured with a regex pattern.  The replacement
+	can be an empty string.  See the following description of
+	``scope=...`` flag about how this is useful.
+
+``scope=ref|push|pop|clear|set``
+
+	Specify what to do with the internal scope stack.
+
+	A parser programmed with ``--regex-<LANG>`` has a stack (scope
+	stack) internally. You can use it for tracking scope
+	information. The ``scope=...`` flag is for manipulating and
+	utilizing the scope stack.
+
+	If ``{scope=push}`` is specified, a tag captured with
+	``--regex-<LANG>`` is pushed to the stack. ``{scope=push}``
+	implies ``{scope=ref}``.
+
+	You can fill the scope field of captured tag with
+	``{scope=ref}``. If ``{scope=ref}`` flag is given,
+	@CTAGS_NAME_EXECUTABLE@ attaches the tag at the top to the tag
+	captured with ``--regex-<LANG>`` as the value for the ``scope:``
+	field.
+
+	@CTAGS_NAME_EXECUTABLE@ pops the tag at the top of the stack when
+	``--regex-<LANG>`` with ``{scope=pop}`` is matched to the input
+	line.
+
+	Specifying ``{scope=clear}`` removes all the tags in the scope.
+	Specifying ``{scope=set}`` removes all the tags in the scope, and
+	then pushes the captured tag as ``{scope=push}`` does.
+
+	In some case, you may want to use ``--regex-<LANG>`` only for side
+	effect: using only for manipulating the stack but not for
+	capturing a tag. In such case, make *replacement* component of
+	``--regex-<LANG>`` option empty with specifying ``{placeholder}``
+	as a regex flag. For an example, you can put a non-named tag to
+	the stack by giving a regex flag ``{scope=push}{placeholder}``.
```

Comment:
"For example, a non-named tag can be put on the stack using the regex flag ``{scope=push}{placeholder}``."